src/main/java/org/mastodon/mamut/feature/branch/BranchSpotFeatureUtils.java

```diff
@@ -146,4 +150,84 @@ public static double[] getLastSpotCoordinates( final Model model, final BranchSp
 		model.getGraph().releaseRef( ref );
 		return lastCoordinates;
 	}
+
+	/**
+	 * Returns the number of leaves of a branch spot.
+	 * <p>
+	 * A leaf is a branch spot that has no children.
+	 * </p>
+	 * Example:
+	 * <pre>
+	 *                      branchSpotA
+	 * 	       ┌──────────────┴─────────────────┐
+	 * 	       │                                │
+	 * 	   branchSpotB                      branchSpotC
+	 * 	                                 ┌──────┴───────┐
+	 * 	                                 │              │
+	 * 	                             branchSpotD branchSpotE
+	 * </pre>
+	 * In this example,
+	 * <ul>
+	 *     <li>branchSpotA has 3 leaves (branchSpotC, branchSpotD, branchSpotE)</li>
+	 *     <li>branchSpotB has 1 leaf (branchSpotB)</li>
+	 *     <li>branchSpotC has 2 leaves (branchSpotD, branchSpotE)</li>
+	 *     <li>branchSpotD has 1 leaf (branchSpotD)</li>
+	 *     <li>branchSpotE has 1 leaf (branchSpotE)</li>
+	 * </ul>
+	 *
+	 * @param branchGraph the model branch graph, which contains the branch spot
+	 * @param branchSpot the branch spot
+	 * @return the number of leaves of the branch spot
+	 */
+	public static int countLeaves( final ModelBranchGraph branchGraph, final BranchSpot branchSpot )
+	{
+		int n = 0;
+		for ( DepthFirstIteration.Step< BranchSpot > step : DepthFirstIteration.forRoot( branchGraph, branchSpot ) )
+		{
+			if ( step.isLeaf() )
+				n++;
+		}
+		return n;
+	}
+
+	/**
+	 * Returns the total duration of the given branch spot and all its successors.
+	 * <p>
+	 * A leaf is a branch spot that has no children.
+	 * </p>
+	 * Example:
+	 * <pre>
+	 *                  branchSpotA (duration=3)
+	 * 	       ┌──────────────┴─────────────┐
+	 * 	       │                            │
+	 * 	 branchSpotB (duration=2)     branchSpotC (duration=1)
+	 * 	                             ┌──────┴───────────────────┐
+	 * 	                             │                          │
+	 * 	                       branchSpotD (duration=2)   branchSpotE (duration=3)
+	 * </pre>
+	 * In this example, the total duration of the branch spot and all its successors is:
+	 * <ul>
+	 *     <li>branchSpotA = 11 (3+2+1+2+3)</li>
+	 *     <li>branchSpotB = 2 (2)</li>
+	 *     <li>branchSpotC = 6 (1+2+3)</li>
+	 *     <li>branchSpotD = 2 (2)</li>
+	 *     <li>branchSpotE = 3 (3)</li>
+	 * </ul>
+	 * @param branchGraph the model branch graph, which contains the branch spot
+	 * @param branchSpot the branch spot
+	 * @return the total duration of the branch spot and all its successors
+	 */
+	public static int totalBranchDurations( final ModelBranchGraph branchGraph, final BranchSpot branchSpot )
+	{
+		int totalDuration = 0;
+		DepthFirstIterator< BranchSpot, BranchLink > it = new DepthFirstIterator<>( branchGraph );
+		it.reset( branchSpot );
+		while ( it.hasNext() )
+		{
+			BranchSpot node = it.next();
+			int branchDuration = node.getTimepoint() - node.getFirstTimePoint();
```

Comment:
```suggestion
			int branchDuration = node.getTimepoint() - node.getFirstTimePoint() + 1;
```
A "+1" is needed here, to make things more consistent. And it will also avoid division by zero exceptions. Here is why the "+1" is needed.
1. A singleton spot would have a duration of 1. Which would be equivalent of saying the branch spot exits of the shortest possible amount of time.
2. Summing the duration along a path would make more sense. The following would for example be true: ```branchSpotE.getLastTimepoint() - branchSpotA.getFirstTimepoint() + 1 == duration(branchSpotA) + duration(branchSpotB) + duration(branchSpotC)```